src/stdatamodels/jwst/datamodels/multispec.py

```diff
@@ -57,3 +61,17 @@ def __init__(self, init=None, **kwargs):
             return
 
         super(MultiSpecModel, self).__init__(init=init, **kwargs)
+
+        try:
+            if isinstance(init, fits.hdu.hdulist.HDUList):
```

Comment:
I agree in principle, and hopefully @braingram has a better idea of how to accomplish this. But I don't see a method defined that connects a schema-defined table with unit designations, and I'm unsure of how to even start that. Even just saving an existing MultiSpecModel from fits to asdf, then loading the asdf gives different data structure types for the table - the asdf file returns `<class 'asdf.tags.core.ndarray.NDArrayType'>` with no column attributes, while the fits version returns `<class 'astropy.io.fits.fitsrec.FITS_rec'>`. I don't know enough about the asdf ndarray to understand if it allows units, but the FITS_rec does allow this, with the downside that it stores that data in the FITS keywords...

Reply:
FWIW, @nden pointed me to the unit schemas in asdf that may or may not be used in Roman datamodels, but I don't know if/how they're being implemented for tables.

Reply:
Thanks for putting this together. I should hopefully have a chance to look into this PR tomorrow in more depth. One question I have is: does make sense to put the units in the schema? There are a few other schemas that have units (here's one [example](https://github.com/spacetelescope/stdatamodels/blob/12a4c3acf844c4121fbed4c2b6fb3bf6c04b5039/src/stdatamodels/jwst/datamodels/schemas/msatargacq.schema.yaml#L36)). I am not sure if or how the units in the schema are used. It might be possible to have stdatamodels use the units in the schema during write to define the `TUINT` headers and on read to fill in the units from the schema (perhaps instead of or only if there are no `TUNIT` headers).

Reply:
The units here are variable, which makes me think that example would be hard to mimic - I tried to pull the unit schema from asdf into the schema directory here and enter something like:
```
- name: WAVELENGTH
  datatype: float64
  unit:
    - ref: unit.schema
```

in the spectable.schema.yaml file, but from what I can tell, datamodels does not have any method of storing/validating attributes of a table/array beyond name and datatype.

Reply:
Should I close this PR and have a new one for your fixes, or should I just commit those into this PR and remove my edits?

Reply:
It may be easiest if I open a new one. I sent the link to @jemorrison and if the new PR doesn't fix the issue (but this one does) then I'll close the new PR.